### DISCUSSION TITLE
Public Enderman Farm

### DISCUSSION THREAD

**Discussion by aisukuma** [Stardust-Labs-MC/StardustMC#40] [Ideas]
Who wants to be responsible to build **ONE** public enderman farm in the end dimension? While there are other types of XP farms out there and XP farms that grant way more XP than that of an enderman farm, it would be nice to have one for the public.

**Comment by g-g-glitch** [comment]
I'm definitely very much interested in this build. I have a solid Enderman farm based on iJevin's design with excellent efficiency on vanilla Minecraft. (Schematic attached.) 

👉 [Glitch_EndermanFarm_v0.1.litematic.zip](https://github.com/Stardust-Labs-MC/StardustMC/files/12474861/Glitch_EndermanFarm_v0.1.litematic.zip)

While I don't consider this design finalized, functionally it should be solid — efficient with spawn prevention and teleportation prevention on surfaces outside of the farm's internals.  Aesthetically, I want to make further improvements — perhaps changing out the block pallet for endstone and purpur and modifying the overall structure to be more grand.

**Design notes:**

With regard to preventing enderman teleportation, double-carpets with gaps visible from the side are strictly avoided for improved aesthetics. In such areas, waterlogged blocks have been used instead. (I've tested these surfaces while aggro'ing endermen in survival mode and it worked successfully.)

- **Enderman spawning** in unwanted locations is prevented by having those areas with a light level of one or greater.
- **Enderman teleportation** to unwanted locations is prevented by varying methods:
	- Teleportation onto the walkway is prevented by two methods:
		1. Walkway sides have **waterlogged blocks**.
		2. Walkway middle has **carpet covered torches**.
			- Note: double-carpet would thwart teleportation, but the light from the torches also prevents spawning.
	- Teleportation into player rooms is prevented by having the **ceiling slightly less than three blocks high**.
		- Note: waterlogged blocks could be used instead, if we want to raise the ceiling height.
	- Teleportation onto the roof of player rooms is prevented by having **waterlogged stairs and slabs**.

In general, whether we use a variation of this design or another, I think the enderman farm should meet the following requirements:

- **Functional requirements:**
    - Performs with high efficiency.
    - Damages endermen for one-hit kills.
    - Does not allow enderman teleportation to undesirable locations.
    - Does not allow enderman spawning in undesirable locations.
- **Aesthetic requirements:**
    - Has no carpet air gaps visible from the side.

**Comment by CodingPanda0623** [comment]
I believe the recent solution to endermen teleporting away was to use waterlogged leaves, as they don’t flow all over the place but still keep endermen from teleporting on them. I was actually planning on building an enderman farm that was quite efficient but unfortunately needs a good sword to work with. I believe there was another design that I’ll look into where it uses the player to aggro the endermen, sweeping edge on an armor stand to damage them, and they eventually die of entity cramming.

**Reply by g-g-glitch** [reply]
I am well aware of waterlogged leaves, and I am intentionally avoiding that aesthetic. If waterlogged leaves are used, the rest of the build should aesthetically match. But, again, I’m not going for that aesthetic. 

Done right, water won't "flow all over the place" with waterlogged stairs and slabs. (Having said that, artistically tasteful waterfalls into the void can be quite beautiful, if that is part of the visual intention.)

My two big aesthetic pet peeves with many builds in the End are (1) ugly, visually clashing, waterlogged leaves and (2) ugly air gaps between double-carpets (there methods to conceal those air gaps).

**Comment by aisukuma** [comment]
We now have a public enderman thanks to G_GLITCH :)
